replicaset.go

```diff
@@ -66,25 +77,39 @@ func Initiate(session *mgo.Session, address, name string, tags map[string]string
 	monotonicSession := session.Clone()
 	defer monotonicSession.Close()
 	monotonicSession.SetMode(mgo.Monotonic, true)
-	cfg := Config{
-		Name:    name,
-		Version: 1,
-		Members: []Member{{
-			Id:      1,
-			Address: fixIpv6Address(address),
-			Tags:    tags,
-		}},
-	}
-	logger.Infof("Initiating replicaset with config %#v", cfg)
+	// We don't know mongod's ability to use a correct IPv6 addr format
+	// until the server is started, but we need to know before we can start
+	// it. Try the older, incorrect format, if the correct format fails.
+	cfg := []Config{
+		Config{
+			Name:    name,
+			Version: 1,
+			Members: []Member{{
+				Id:      1,
+				Address: address,
+				Tags:    tags,
+			}},
+		},
+		Config{
+			Name:    name,
+			Version: 1,
+			Members: []Member{{
+				Id:      1,
+				Address: formatIPv6AddressWithoutBrackets(address),
+				Tags:    tags,
+			}},
+		},
+	}
+
 	var err error
+	// Attempt replSetInitiate, with potential retries.
 	for i := 0; i < maxInitiateAttempts; i++ {
 		monotonicSession.Refresh()
-		err = monotonicSession.Run(bson.D{{"replSetInitiate", cfg}}, nil)
-		if err != nil && err.Error() == rsMembersUnreachableError {
+		err = attemptInitiate(monotonicSession, cfg)
+		if err != nil {
```

Comment:
I think this becomes
if err != nil {
  time.Sleep()
} else {
  break
}

Otherwise this is triggering maxInitiateAttempts always.